apps/web/src/ee/clerk/providers/EnterpriseAuthProvider.tsx

```diff
@@ -0,0 +1,189 @@
+  inPublicRoute: boolean;
+  inPrivateRoute: boolean;
+  isLoading: boolean;
+  currentUser: IUserEntity | undefined;
+  organizations: IOrganizationEntity[];
+  currentOrganization: IOrganizationEntity | undefined;
+  login: (...args: any[]) => void;
+  logout: () => void;
+  environmentId: string | null;
+}
+
+const toUserEntity = (clerkUser: UserResource): IUserEntity => ({
+  _id: clerkUser.id,
+  firstName: clerkUser.firstName,
+  lastName: clerkUser.lastName,
+  email: clerkUser.emailAddresses[0].emailAddress,
+  profilePicture: clerkUser.imageUrl,
+  createdAt: clerkUser.createdAt?.toString() ?? '',
+  showOnBoarding: clerkUser.publicMetadata.showOnBoarding,
+  showOnBoardingTour: clerkUser.publicMetadata.showOnBoardingTour,
+  servicesHashes: clerkUser.publicMetadata.servicesHashes,
+  jobTitle: clerkUser.publicMetadata.jobTitle,
+  hasPassword: clerkUser.passwordEnabled,
+});
+
+const toOrganizationEntity = (clerkOrganization: OrganizationResource): IOrganizationEntity => ({
+  _id: clerkOrganization.id,
+  name: clerkOrganization.name,
+  createdAt: clerkOrganization.createdAt.toString(),
+  updatedAt: clerkOrganization.updatedAt.toString(),
+  apiServiceLevel: clerkOrganization.publicMetadata.apiServiceLevel,
+  defaultLocale: clerkOrganization.publicMetadata.defaultLocale,
+  domain: clerkOrganization.publicMetadata.domain,
+  productUseCases: clerkOrganization.publicMetadata.productUseCases,
+});
+
+const AuthEnterpriseContext = createContext<AuthEnterpriseContextProps | undefined>(undefined);
+
+export const useAuthEnterpriseContext = () => {
+  const context = useContext(AuthEnterpriseContext);
+  if (context === undefined) {
+    throw new Error('useAuthEnterpriseContext must be used within an AuthEnterpriseProvider');
+  }
+
+  return context;
+};
+
+const _AuthEnterpriseProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const { signOut, userId, orgId } = useAuth();
+  const { user: clerkUser, isLoaded: isUserLoaded } = useUser();
+  const { organization: clerkOrganization, isLoaded: isOrganizationLoaded } = useOrganization();
+  const { setActive, isLoaded: isOrgListLoaded } = useOrganizationList();
+
+  const [user, setUser] = useState<IUserEntity | undefined>(undefined);
+  const [organization, setOrganization] = useState<IOrganizationEntity | undefined>(undefined);
+
+  const ldClient = useLDClient();
+  const segment = useSegment();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const inPublicRoute = useMemo(
+    () => Array.from(PUBLIC_ROUTES_PREFIXES.values()).some((prefix) => location.pathname.startsWith(prefix)),
+    [location.pathname]
+  );
+  const inPrivateRoute = !inPublicRoute;
+
+  const logout = useCallback(() => {
+    queryClient.clear();
+    segment.reset();
+    navigate(ROUTES.AUTH_LOGIN);
+    signOut();
+  }, [navigate, queryClient, segment, signOut]);
+
+  useEffect(() => {
+    if (orgId || !userId) return;
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Why this check?‏

Reply:
- removed the `userId` one cause its redundant
- `orgId` is used to check if user has already active organization, in such case we dont need to choose any